### PR TITLE
Giant macro pad

### PR DESCRIPTION
**Pull request by mini-ninja-64**
## Description

This PR will add a 20x20 macropad with a default & via keymap

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

None

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/cutie_club/giant_macro_pad/keymaps/via/config.h [hunk 1]
```diff
@@ -0,0 +1 @@
+#define DYNAMIC_KEYMAP_LAYER_COUNT 1
```
[the remark this answers is not in the document]
Comment (mini-ninja-64) [reply keyboards/cutie_club/giant_macro_pad/keymaps/via/config.h:19]: Thankyou, I have rectified the issue 😄

File: keyboards/cutie_club/giant_macro_pad/chconf.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+#define CH_CFG_USE_WAITEXIT TRUE
+
+#define CH_CFG_USE_CONDVARS TRUE
+
+#define CH_CFG_USE_CONDVARS_TIMEOUT FALSE
+
+#define CH_CFG_USE_MESSAGES TRUE
+
+#define CH_CFG_USE_MAILBOXES TRUE
```
Comment (tzarc) [inline keyboards/cutie_club/giant_macro_pad/chconf.h:33]: Can you explain why you need these? Is this file needed?

File: keyboards/cutie_club/giant_macro_pad/mcuconf.h [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include_next <mcuconf.h>
+
+#undef STM32_PWM_USE_TIM3
+#define STM32_PWM_USE_TIM3 TRUE
```
Comment (tzarc) [inline keyboards/cutie_club/giant_macro_pad/mcuconf.h:22]: Can you explain why you need this? Is this file needed?

File: keyboards/cutie_club/giant_macro_pad/readme.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+# Cupar19 Giant Macro Pad
+
+![Cupar19 Giant Macro Pad](https://user-images.githubusercontent.com/15253132/118325863-0135e680-b4fc-11eb-99f2-754b155b0acd.png)
+
+* Keyboard Maintainer: [Cutie Club](https://github.com/cutie-club/)
+* Hardware Supported: STM32F072 based giant 20x20 macro pad
```
Comment (tzarc) [inline keyboards/cutie_club/giant_macro_pad/readme.md:6]: If this is a one-off, shouldn't it go into `keyboards/handwired/`?

**Review by tzarc — CHANGES_REQUESTED** [review]
Some queries.

File: keyboards/cutie_club/giant_macro_pad/readme.md (hunk 4 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/giant_macro_pad/readme.md:6]: Thank you for the response

"One off" is bad wording on my part, 5 exists, It just wasnt a group buy or anything, but a PCB commission.

I feel it would not fit in handwired so well since its not handwired?

What do you propose, I put instead for Hardware Availability? or maybe it would fit in the Handwired folder?

File: keyboards/cutie_club/giant_macro_pad/chconf.h (hunk 2 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/giant_macro_pad/chconf.h:33]: Many apologies, my mistake, I do not think I need this file at all 😄 

I will delete them and mark this conversation as resolved

File: keyboards/cutie_club/giant_macro_pad/mcuconf.h (hunk 3 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/giant_macro_pad/mcuconf.h:22]: I do not think I need this file at all 😄 

I will delete them and mark this conversation as resolved

File: keyboards/cutie_club/giant_macro_pad/chconf.h (hunk 2 above)
Comment (mini-ninja-64) [reply keyboards/cutie_club/giant_macro_pad/chconf.h]: Presuming ST_Frequency etc is now a QMK default based on MCU?

Would I need to specify a generic BOARD or is this handled by the build system?

File: keyboards/cutie_club/giant_macro_pad/readme.md (hunk 4 above)
Comment (tzarc) [reply keyboards/cutie_club/giant_macro_pad/readme.md:6]: All good for now, will raise some discussions with the other collaborators regarding this sort of thing.

File: keyboards/cutie_club/giant_macro_pad/chconf.h (hunk 2 above)
Comment (tzarc) [reply keyboards/cutie_club/giant_macro_pad/chconf.h]: Defaults live under `platforms/chibios/common/configs` and `platforms/chibios/GENERIC_STM32_F072XB/configs`. What you'd done is override the defaults, asking for extra things from ChibiOS... which you weren't using.

Comment (tzarc) [reply keyboards/cutie_club/giant_macro_pad/chconf.h]: Main intention is that the defaults are present so that under most circumstances designers don't need to modify anything apart from the `MCU = xxxxxx` line in the rules.mk file.

**Review by tzarc — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]
Thanks!
